compiler/rustc_data_structures/src/memmap.rs

```diff
@@ -19,7 +19,7 @@ impl Mmap {
     #[inline]
     pub unsafe fn map(file: File) -> io::Result<Self> {
         // Safety: the caller must ensure that this is safe.
-        unsafe { memmap2::Mmap::map(&file).map(Mmap) }
+        unsafe { memmap2::MmapOptions::new().map_copy_read_only(&file).map(Mmap) }
```

Comment:
Can you add a comment that explains that the default `MmapOptions` will pass though updates to the file, wheras this setting *might* pass through updates, but we really prefer the updates-are-optional version because it is more widely supported.

Reply:
Will do.

Reply:
You may now check whether or not the modifications to the PR made (git commit message and comment) are suitable.

In my opinion, the comment is too long, but I'm a bit at a loss to make it shorter. Please advise.

Reply:
@rustbot review

Reply:
This comment should be in the function body as a `//` comment, not as a doc comment with `///` outside, because it describes an implementation detail not the API.

Your comment is long because it's too specific. `memmap2` supports Windows, so talking about flags to the `mmap` syscall is speaking to the wrong layer of abstraction here. Instead, say something like:
```
// The default behavior of `memmap2` is to map such that updates to the file
// are visible in the mapping. We don't want updates, so all we're doing
// here is turning off an unused feature.
```

For the second paragraph I'd just say something like
```
// Most importantly, this `memmap2` call is more widely-supported: https://github.com/rust-lang/rust/issues/122262
```
Your explanation in the issue is quite thorough, and I wouldn't bother hurting yourself by trying to cram too much of it into a comment. We're turning off an unused feature to work on more systems. That's good enough for a casual reader.

Reply:
I tried shortening it.

(Sorry for the delay: my dev machine is precisely the one being virtualised on which I have all these problems here.)

@rustbot review

Reply:
Rather than slowly going back and forth here, I'm going to reword the comment in a way that I think we'd both be happy with and approve this.

Reply:
Wow. Today's developments are even more impressive in terms of intolerance on #irc. This is definitely a community to stay away from.

Reply:
What IRC are you on? If you have trouble with any of the official Rust spaces, please contact the moderation team: https://www.rust-lang.org/governance/teams/moderation

Reply:
I'm not going through official complaint procedures. These people banned me. They are free to behave like abusive -------- for petty reasons. I'm just making it known that I also have limits to my tolerance. I hope this is my last message on anything related officially to rust.

Please let that be the last message on this thread: it's already polluted enough, and has been as soon as I mentioned in the upstream issue that people on irc were opposed to the change I proposed.

In two weeks involved with rust, I have met only opposition and scorn wrapped in fake netiquette. I'm done.

The language is really cool and something I'd recommend. The people, however, are not. At times, being nonjudgemental is nothing else than leniency, and this, for me, would be such a time.

Let's all drop it. Final message.

And feel free to revert that commit if any heat comes that way.